src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -167,7 +167,10 @@ class ClusterMonitorActor(val cluster: Cluster,
       gdDAO.deleteCluster(cluster.googleProject, cluster.clusterName),
       // Remove the Dataproc Worker IAM role for the pet service account
       // Only do this if the cluster was created with the pet service account.
-      removeIamRolesForUser
+      removeIamRolesForUser,
+      // Remove the service account key in Google, if present.
+      // Only do this if the cluster was NOT created with the pet service account.
```

Comment:
"only do this if" = "this only happens if"? or "only do this if" = "i should have put an if statement here but i forgot"?

Reply:
“Only happens if” sounds like better wording.. will change this.